tiny_dnn/util/gradient_check.h

```diff
@@ -52,8 +52,8 @@ float_t numeric_gradient(layer &layer,
                          std::vector<tensor_t> in_data,  //  copy is safer
                          const size_t in_edge,
                          const size_t in_pos,
-                         std::vector<tensor_t> &out_data,
-                         std::vector<tensor_t> &out_grads,
+                         std::vector<tensor_t> out_data,
```

Comment:
Because the same out grads might be used in other tests and we don't want this function to modify them. Also, since this function is only called on tests, it is computationally safe to use a pass by value here.